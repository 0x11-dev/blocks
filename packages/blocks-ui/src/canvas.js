/** @jsx jsx */
import React, { useState } from 'react'
import { jsx, Styled } from 'theme-ui'
import prettier from 'prettier/standalone'
import parserJS from 'prettier/parser-babylon'

import { useEditor } from './editor-context'
import InlineRender from './inline-render'

import { Clipboard, Check } from 'react-feather'
import { IconButton } from './ui'

const Wrap = props => (
  <div
    sx={{
      position: 'relative',
      width: '60%',
      backgroundColor: 'white',
      height: 'calc(100vh - 43px)',
      overflow: 'auto'
    }}
    {...props}
  />
)

function copyToClipboard(toCopy) {
  const el = document.createElement(`textarea`)
  el.value = toCopy
  el.setAttribute(`readonly`, ``)
  el.style.position = `absolute`
  el.style.left = `-9999px`
  document.body.appendChild(el)
  el.select()
  document.execCommand(`copy`)
  document.body.removeChild(el)
}

const Copy = ({ toCopy }) => {
  const [hasCopied, setHasCopied] = useState(false)

  function copyToClipboardOnClick() {
    if (hasCopied) return

    copyToClipboard(toCopy)
    setHasCopied(true)

    setTimeout(() => {
      setHasCopied(false)
    }, 2000)
  }

  return (
    <IconButton
      onClick={copyToClipboardOnClick}
      sx={{ position: 'absolute', right: 0 }}
    >
      {hasCopied ? (
        <Check sx={{ color: 'green' }} aria-label="Copied" />
      ) : (
        <Clipboard aria-label="Copy" />
      )}
    </IconButton>
  )
}

export default ({ code, transformedCode, scope, theme }) => {
  const { mode } = useEditor()
  const formattedCode = prettier.format(code, {
    parser: 'babel',
    plugins: [parserJS]
  })

  if (mode === 'code') {
    return (
      <Wrap>
        <Copy toCopy={formattedCode} />
        <Styled.pre
          language="js"
          sx={{
            mt: 0,
            backgroundColor: 'white',
            color: 'black'
          }}
        >
          {formattedCode}
        </Styled.pre>
      </Wrap>
    )
  }

  return (
    <Wrap>
      <InlineRender scope={scope} code={transformedCode} theme={theme} />
    </Wrap>
  )
}
