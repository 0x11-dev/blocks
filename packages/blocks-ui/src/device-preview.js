/** @jsx jsx */
import { jsx } from 'theme-ui'
import {
  Children,
  cloneElement,
  useLayoutEffect,
  useRef,
  useState
} from 'react'
import { createPortal } from 'react-dom'
import { ZoomIn, ZoomOut } from 'react-feather'
import { CacheProvider, Global } from '@emotion/core'
import createCache from '@emotion/cache'
import weakMemoize from '@emotion/weak-memoize'

import { IconButton } from './ui'

const MIN_ZOOM_LEVEL = 25

const createCacheWithContainer = weakMemoize(container =>
  createCache({ container })
)

function Frame({ children, ...restProps }) {
  const frameRef = useRef()
  const [[head, body], setNodes] = useState([])
  useLayoutEffect(() => {
    const { contentDocument } = frameRef.current
    setNodes([contentDocument.head, contentDocument.body])
  }, [])
  return (
    <iframe ref={frameRef} {...restProps}>
      {body
        ? createPortal(
            <CacheProvider value={createCacheWithContainer(head)}>
              <Global styles={{ body: { margin: 0 } }} />
              {children}
            </CacheProvider>,
            body
          )
        : null}
    </iframe>
  )
}

export const Device = ({ children, width, height, name, zoomLevel }) => {
  return (
    <div css={{ flex: '0 0 auto', margin: 16 }}>
      <div css={{ flex: '1 1 auto' }}>
        <div
          css={{
            fontSize: 14,
            textAlign: 'left',
            marginBottom: 8,
            color: '#444'
          }}
        >
          <strong>{name}:</strong> {width}px
        </div>
        <div
          style={{
            width: width * zoomLevel,
            height: height * zoomLevel,
            boxShadow: '0 2px 8px rgba(0, 0, 0, 0.2)'
          }}
        >
          <Frame
            css={{
              margin: 0,
              border: 0,
              transformOrigin: `top left`
            }}
            style={{
              width: width,
              height: height,
              transform: `scale(${zoomLevel})`
            }}
          >
            {children}
          </Frame>
        </div>
      </div>
    </div>
  )
}

export function PreviewArea({ children }) {
  const [wrap, setWrap] = useState(false)
  const [zoomLevel, setZoomLevel] = useState(1)
  const shiftKeyDown = useRef()
  return (
    <div
      css={{
        display: 'grid',
        gridTemplateRows: 'auto 1fr',
        width: '60%',
        height: '100%',
        overflow: 'auto'
      }}
    >
      <div
        css={{
          display: 'grid',
          gridAutoFlow: 'column',
          gridAutoColumns: 'minmax(min-content, max-content)',
          gridGap: 8,
          alignItems: 'center',
          padding: 16
        }}
      >
        <input
          type="number"
          min={MIN_ZOOM_LEVEL}
          value={(zoomLevel * 100).toFixed(0)}
          onKeyDown={event => (shiftKeyDown.current = event.shiftKey)}
          onChange={event => {
            const nextZoomLevel = parseFloat(event.target.value) / 100
            const shiftAmount = nextZoomLevel < zoomLevel ? -0.1 : 0.1
            setZoomLevel(
              nextZoomLevel + (shiftKeyDown.current ? shiftAmount : 0)
            )
          }}
          css={{
            appearance: 'none',
            WebkitAppearance: 'none',
            width: 30,
            fontSize: 16,
            border: 'none',
            backgroundColor: 'transparent',
            '&::-webkit-inner-spin-button, &::-webkit-outer-spin-button': {
              WebkitAppearance: 'none',
              margin: 0
            }
          }}
        />
        <span>%</span>
        <IconButton
          aria-label="Zoom viewport out"
          onClick={() => {
            const nextZoomLevel = zoomLevel - 0.1
            if (nextZoomLevel > 0.25) {
              setZoomLevel(nextZoomLevel)
            }
          }}
        >
          <ZoomOut size={15} sx={{ position: 'relative', top: '1px' }} />
        </IconButton>
        <input
          type="range"
          min={MIN_ZOOM_LEVEL}
          max={200}
          value={(zoomLevel * 100).toFixed(0)}
          onChange={event => setZoomLevel(parseFloat(event.target.value) / 100)}
          style={{ width: 80 }}
        />
        <IconButton
          aria-label="Zoom viewport in"
          onClick={() => setZoomLevel(zoomLevel + 0.1)}
        >
          <ZoomIn size={15} sx={{ position: 'relative', top: '1px' }} />
        </IconButton>
        <label
          css={{
            display: 'flex',
            alignItems: 'center',
            userSelect: 'none'
          }}
        >
          <input
            type="checkbox"
            checked={wrap}
            onChange={event => setWrap(event.target.checked)}
          />
          <span style={{ fontSize: 14 }}>wrap</span>
        </label>
      </div>
      <div
        css={{ display: 'flex', overflow: 'auto', backgroundColor: '#f0f0f0' }}
      >
        <div
          css={{
            display: 'flex',
            flexWrap: wrap ? 'wrap' : undefined,
            alignItems: 'start',
            alignContent: 'start',
            height: '100%',
            padding: 16
          }}
        >
          {Children.map(children, child => cloneElement(child, { zoomLevel }))}
        </div>
      </div>
    </div>
  )
}
