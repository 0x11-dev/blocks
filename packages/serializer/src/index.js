import unified from 'unified'
import mdx from 'remark-mdx'
import parse from 'remark-parse'
import stringify from 'remark-stringify'
import squeezeParagraphs from 'remark-squeeze-paragraphs'
import visit from 'unist-util-visit'
import { Data, Value } from 'slate'

import slateCompiler from './remark-slate'
import interleave from './remark-interleave'
import { getComponentName, toJS } from './util'
import { parseJSXBlock, applyProps } from './parse-jsx'
import { Visitor } from 'handlebars'

const parser = unified()
  .use(parse, {
    commonmark: true,
    position: false
  })
  .use(squeezeParagraphs)
  .use(interleave)
  .use(mdx)

export const parseMDX = md => parser.runSync(parser.parse(md))

const stringifier = unified()
  .use(stringify, {
    bullet: '*',
    fences: true
  })
  .use(squeezeParagraphs)
  .use(mdx)

export const stringifyMDX = mdxast =>
  stringifier.stringify(stringifier.runSync(mdxast))

// Turn an MDX string into Slate schema
export const deserialize = mdx => {
  const result = parser.use(slateCompiler).processSync(mdx)

  return Value.fromJSON(result.contents)
}
