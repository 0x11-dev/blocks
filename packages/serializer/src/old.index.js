import unified from 'unified'
import mdx from 'remark-mdx'
import parse from 'remark-parse'
import stringify from 'remark-stringify'
import squeezeParagraphs from 'remark-squeeze-paragraphs'
import { Data } from 'slate'

import interleave from './remark-interleave'
import { getComponentName, toJS } from './util'
import { parseJSXBlock, applyProps } from './parse-jsx'

const parser = unified()
  .use(parse, {
    commonmark: true,
    position: false
  })
  .use(squeezeParagraphs)
  .use(interleave)
  .use(mdx)

export const parseMDX = md => parser.runSync(parser.parse(md))

const stringifier = unified()
  .use(stringify, {
    bullet: '*',
    fences: true
  })
  .use(squeezeParagraphs)
  .use(mdx)

export const stringifyMDX = mdxast =>
  stringifier.stringify(stringifier.runSync(mdxast))

const paragraph = {
  match: node => node.object === 'block' && node.type === 'paragraph',
  matchMdast: node => node.type === 'paragraph',
  fromMdast: (node, _index, parent, { visitChildren }) => {
    return {
      object: 'block',
      type: 'paragraph',
      nodes: visitChildren(node)
    }
  },
  toMdast: (object, _index, parent, { visitChildren }) => {
    return {
      type: 'paragraph',
      children: visitChildren(object)
    }
  }
}

const br = {
  match: node => node.type === 'break',
  matchMdast: node => node.type === 'break',
  fromMdast: (node, _index, _parent, { visitChildren }) => {
    return {
      object: 'text',
      leaves: [
        {
          object: 'leaf',
          text: '\n'
        }
      ]
    }
  }
}

const image = {
  match: node => node.object === 'block' && node.type === 'image',
  matchMdast: node => node.type === 'image',
  fromMdast: node => {
    return {
      object: 'block',
      type: 'image',
      isVoid: true,
      data: {
        alt: node.alt,
        src: node.url
      },
      nodes: []
    }
  },
  toMdast: object => {
    return {
      type: 'image',
      alt: object.data.alt,
      url: object.data.src
    }
  }
}

const blockQuote = {
  match: node => node.object === 'block' && node.type === 'block-quote',
  matchMdast: node => node.type === 'blockquote',
  fromMdast: (node, index, parent, { visitChildren }) => ({
    object: 'block',
    type: 'block-quote',
    nodes: visitChildren(node)
  }),
  toMdast: (object, index, parent, { visitChildren }) => ({
    type: 'blockquote',
    children: visitChildren(object)
  })
}

const bulletedList = {
  match: node => node.object === 'block' && node.type === 'bulleted-list',
  matchMdast: node => node.type === 'list' && !node.ordered,
  fromMdast: (node, _index, _parent, { visitChildren }) => ({
    object: 'block',
    type: 'bulleted-list',
    nodes: visitChildren(node)
  }),
  toMdast: (object, _index, _parent, { visitChildren }) => {
    return {
      type: 'list',
      ordered: false,
      children: visitChildren(object)
    }
  }
}

const numberedList = {
  match: node => node.object === 'block' && node.type === 'numbered-list',
  matchMdast: node => node.type === 'list' && node.ordered,
  fromMdast: (node, _index, _parent, { visitChildren }) => ({
    object: 'block',
    type: 'numbered-list',
    nodes: visitChildren(node)
  }),
  toMdast: (object, _index, _parent, { visitChildren }) => {
    return {
      type: 'list',
      ordered: true,
      children: visitChildren(object)
    }
  }
}

const listItem = {
  match: node => node.object === 'block' && node.type === 'list-item',
  matchMdast: node => node.type === 'listItem',
  fromMdast: (node, index, parent, { visitChildren }) => {
    return {
      object: 'block',
      type: 'list-item',
      nodes: visitChildren(node)
    }
  },
  toMdast: (object, _index, _parent, { visitChildren }) => {
    return {
      type: 'listItem',
      children: visitChildren(object)
    }
  }
}

const listItemChild = {
  match: node => node.object === 'block' && node.type === 'list-item-child',
  matchMdast: (node, _index, parent) =>
    node.type === 'paragraph' && parent.type === 'listItem',
  fromMdast: (node, index, parent, { visitChildren }) => {
    return {
      object: 'block',
      type: 'list-item-child',
      nodes: visitChildren(node)
    }
  },
  toMdast: (object, _index, parent, { visitChildren }) => {
    return {
      type: 'paragraph',
      children: visitChildren(object)
    }
  }
}

const headings = [
  'heading-one',
  'heading-two',
  'heading-three',
  'heading-five',
  'heading-six'
]
  .map((nodeType, headingOffset) => {
    return {
      match: node => node.object === 'block' && node.type === nodeType,
      matchMdast: node =>
        node.type === 'heading' && node.depth === headingOffset + 1,
      fromMdast: (node, index, parent, { visitChildren }) => ({
        object: 'block',
        type: nodeType,
        nodes: visitChildren(node)
      }),
      toMdast: (object, index, parent, { visitChildren }) => ({
        type: 'heading',
        depth: headingOffset + 1,
        children: visitChildren(object)
      })
    }
  })
  .reverse()

const bold = {
  match: node => node.object === 'mark' && node.type === 'bold',
  matchMdast: node => node.type === 'strong',
  fromMdast: (node, index, parent, { visitChildren }) => {
    return {
      object: 'mark',
      type: 'bold',
      nodes: visitChildren(node)
    }
  },
  toMdast: (mark, index, parent, { visitChildren }) => {
    return {
      type: 'strong',
      children: visitChildren(mark)
    }
  }
}

const codeBlock = {
  match: node => node.object === 'block' && node.type === 'pre',
  matchMdast: node => node.type === 'code',
  fromMdast: (node, _index, _parent, { visitChildren }) => {
    return {
      object: 'block',
      type: 'pre',
      nodes: [
        {
          object: 'text',
          leaves: [
            {
              object: 'leaf',
              text: node.value
            }
          ]
        }
      ]
    }
  },
  toMdast: (node, _index, _parent, { visitChildren }) => {
    return {
      type: 'code',
      value: visitChildren(node)
        .map(childNode => childNode.value)
        .filter(Boolean)
        .join('\n')
    }
  }
}

const code = {
  match: node => node.object === 'mark' && node.type === 'code',
  matchMdast: node => node.type === 'inlineCode',
  fromMdast: (node, index, parent, { visitChildren }) => {
    return {
      object: 'mark',
      type: 'code',
      nodes: [
        {
          object: 'text',
          leaves: [
            {
              marks: [],
              object: 'leaf',
              text: node.value
            }
          ]
        }
      ]
    }
  },
  toMdast: (mark, index, parent, { visitChildren }) => {
    return {
      type: 'inlineCode',
      value: visitChildren(mark)
        .map(childNode => childNode.value)
        .join('')
    }
  }
}

const italic = {
  match: node => node.object === 'mark' && node.type === 'italic',
  matchMdast: node => node.type === 'emphasis',
  fromMdast: (node, index, parent, { visitChildren }) => ({
    object: 'mark',
    type: 'italic',
    nodes: visitChildren(node)
  }),
  toMdast: (mark, index, parent, { visitChildren }) => ({
    type: 'emphasis',
    children: visitChildren(mark)
  })
}

// Inline doesn't currently work correctly
// What happens is:
// Given input like this:
//  plain text in a paragraph <span>jsx children</span>.
// it thinks that <span> is a jsx node and </span> is a jsx node, but jsx children is not
// Currently, this editor doesn't seem to support inline JSX though.
// rendering this doesn't break anything, it just doesn't really do anything special.
const jsxMark = {
  match: node => {
    return node.type === 'jsx' && node.object === 'mark'
  },
  matchMdast: (node, index, parent) => {
    return node.type === 'jsx' && (!parent || parent.type !== 'root')
  },
  fromMdast: (node, index, parent, { visitChildren, context }) => {
    return {
      object: 'mark',
      type: 'jsx',
      text: node.value,
      nodes: []
    }
  },
  toMdast: (mark, index, parent, { visitChildren }) => {
    return {
      type: 'jsx',
      value: visitChildren(mark)
        .map(childNode => childNode.value)
        .join()
    }
  }
}

const isJSX = node => {
  if (node.object !== 'block') return false
  return node.type === 'jsx' || node.type === 'jsx-void'
}

const jsxBlock = {
  match: isJSX,
  matchMdast: (node, index, parent) =>
    node.type === 'jsx' && parent && parent.type === 'root',
  fromMdast: (node, index, parent, { visitChildren }) => {
    let data = {}
    if (node.children) {
      data = {
        type: getComponentName(node.children[0].value),
        props: {}
      }

      // Remove open and closing jsx blocks
      node.children = node.children.slice(1, node.children.length - 1)

      return {
        object: 'block',
        type: 'jsx',
        nodes: visitChildren(node),
        data: {
          type: data.type,
          props: Data.create(data.props)
        }
      }
    } else {
      data = parseJSXBlock(node.value)
      return {
        object: 'block',
        type: 'jsx-void',
        data: {
          type: data.type,
          props: Data.create(data.props)
        }
      }
    }
  },
  toMdast: (object, index, parent, { visitChildren }) => {
    const props = toJS(object.data.props)

    if (!object.data.type) {
      return {
        type: 'jsx',
        value: visitChildren(object)[0].value
      }
    }

    const value = applyProps(`<${object.data.type} />`, { props })

    if (object.type === 'jsx-void') {
      return {
        type: 'jsx',
        value
      }
    }

    const children = visitChildren(object)
    return [
      {
        type: 'jsx',
        value: value.replace(/ \/>$/, '>')
      },
      ...children,
      {
        type: 'jsx',
        value: `</${object.data.type}>`
      }
    ]
  }
}

const link = {
  match: node => node.object === 'inline' && node.type === 'link',
  matchMdast: node => node.type === 'link',
  fromMdast: (node, index, parent, { visitChildren }) => {
    return {
      object: 'inline',
      type: 'link',
      data: {
        href: node.url,
        title: node.title,
        target: node.target
      },
      nodes: visitChildren(node)
    }
  },
  toMdast: (mark, index, parent, { visitChildren }) => ({
    type: 'link',
    url: mark.data.href,
    title: mark.data.title,
    target: mark.data.target,
    children: visitChildren(mark)
  })
}

export const serializer = new MarkdownSerializer({
  rules: [
    listItemChild, // We want this to run before paragraph because it's a special case
    paragraph,
    //    br,
    bold,
    //    code,
    italic,
    jsxMark,
    blockQuote,
    jsxBlock,
    //    codeBlock,
    image,
    link,
    bulletedList,
    numberedList,
    listItem
  ].concat(headings)
})
