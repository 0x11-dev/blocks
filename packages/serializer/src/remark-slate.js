import flatten from 'lodash.flatten'
import phrasing from 'mdast-util-phrasing'

import { getComponentName } from './util'

const toSlate = node => {
  if (node.type === 'text') {
    return {
      object: 'text',
      text: node.value
    }
  }

  if (node.type === 'heading') {
    return {
      object: 'block',
      type: 'heading-' + node.depth,
      nodes: flatten(node.children.map(toSlate))
    }
  }

  // This is an interleaved JSX block which means an open
  // tag followed by a blank line and then markdown.
  if (node.type === 'jsx' && node.children) {
    // Remove open and closing jsx blocks
    const children = node.children.slice(1, node.children.length - 1)

    return {
      object: 'block',
      type: 'jsx',
      nodes: flatten(children.map(toSlate)),
      data: {
        type: getComponentName(node.children[0].value),
        props: {}
      }
    }
  }

  if (phrasing(node)) {
    const parentMark = {
      object: 'mark',
      type: node.type
    }

    const nodes = node.children.map(child => {
      const childMark = child.type !== 'text' && {
        object: 'mark',
        type: child.type
      }

      return {
        object: 'text',
        text: child.value,
        marks: [parentMark, childMark].filter(Boolean)
      }
    })

    return nodes
  }

  return {
    object: 'block',
    type: node.type,
    nodes: node.children.map(toSlate)
  }
}

export default function remarkSlate() {
  this.Compiler = compiler

  function compiler(node) {
    if (node.type === 'root') {
      console.log(node.children)
      return {
        object: 'value',
        document: {
          object: 'document',
          data: {},
          nodes: node.children.map(toSlate)
        }
      }
    }

    return toSlate(node)
  }
}
