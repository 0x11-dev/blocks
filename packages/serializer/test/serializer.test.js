import { deserialize } from '../src'

const FIXTURE = `
# Hello, __world*!*__

<Block>

## Here's another *heading*

Stuff

</Block>

And more stuff

- apples
- oranges
- potato
`

test('correctly serializes MDX to Slate schema', () => {
  const result = deserialize(FIXTURE)

  expect(result.toJSON()).toMatchSnapshot()
})

// test('correctly passes props in JSX blocks', () => {
//   const result = deserialize('<YouTube id="1234" />')
//
//   expect(result.toJSON()).toMatchSnapshot()
// })
