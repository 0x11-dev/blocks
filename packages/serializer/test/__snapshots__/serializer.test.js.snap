// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly serializes MDX to Slate schema 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Hello, ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "strong",
              },
            ],
            "object": "text",
            "text": "world",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "strong",
              },
              Object {
                "data": Object {},
                "object": "mark",
                "type": "emphasis",
              },
            ],
            "object": "text",
            "text": "",
          },
        ],
        "object": "block",
        "type": "heading-1",
      },
      Object {
        "data": Object {
          "props": Object {},
          "type": "Block",
        },
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "Here's another ",
              },
              Object {
                "marks": Array [
                  Object {
                    "data": Object {},
                    "object": "mark",
                    "type": "emphasis",
                  },
                ],
                "object": "text",
                "text": "heading",
              },
            ],
            "object": "block",
            "type": "heading-2",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "Stuff",
              },
            ],
            "object": "block",
            "type": "paragraph",
          },
        ],
        "object": "block",
        "type": "jsx",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "And more stuff",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "marks": Array [],
                    "object": "text",
                    "text": "apples",
                  },
                ],
                "object": "block",
                "type": "paragraph",
              },
            ],
            "object": "block",
            "type": "listItem",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "marks": Array [],
                    "object": "text",
                    "text": "oranges",
                  },
                ],
                "object": "block",
                "type": "paragraph",
              },
            ],
            "object": "block",
            "type": "listItem",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "marks": Array [],
                    "object": "text",
                    "text": "potato",
                  },
                ],
                "object": "block",
                "type": "paragraph",
              },
            ],
            "object": "block",
            "type": "listItem",
          },
        ],
        "object": "block",
        "type": "list",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;
